xline/src/server/xline_server.rs

```diff
@@ -216,17 +217,20 @@ impl XlineServer {
         reporter
             .set_service_status("", ServingStatus::Serving)
             .await;
-        Ok(Server::builder()
-            .add_service(RpcLockServer::new(lock_server))
-            .add_service(RpcKvServer::new(kv_server))
-            .add_service(RpcLeaseServer::from_arc(lease_server))
-            .add_service(RpcAuthServer::new(auth_server))
-            .add_service(RpcWatchServer::new(watch_server))
-            .add_service(RpcMaintenanceServer::new(maintenance_server))
-            .add_service(ProtocolServer::new(curp_server))
-            .add_service(health_server)
-            .serve(addr)
-            .await?)
+        let _h = tokio::spawn(async move {
```

Comment:
Why do you spawn a task to build `XlineServer`?

Reply:
Let `XlineServer::start` not wait at the await point, then I can wait for the ctrl-c signal and call `XlineServer::stop`